packages/node-xmpp-server/lib/S2S/Router.js

```diff
@@ -51,20 +56,29 @@ Router.prototype.credentials = {} // TLS credentials per domain
 /*
  * little helper, because dealing with crypto & fs gets unwieldy
  */
-Router.prototype.loadCredentials = function (domain, key, cert) {
-  var creds = tls.createSecureContext({
+Router.prototype.loadCredentials = function (domain, key, cert, ca) {
+  var creds = {
     key: key,
     cert: cert
-  })
+  }
 
-  this.getContext(domain).setCredentials(creds)
+  if (ca) {
+    creds.ca = ca
+  }
```

Comment:
Looks like it's safe to remove these checks. they are done in `tls.createSecureContext`

https://github.com/nodejs/node/blob/dcccbfdc799d174901c29d82874457367231fec5/lib/_tls_common.js#L51